zerver/openapi/zulip.yaml

```diff
@@ -4646,19 +4646,22 @@ components:
       name: narrow
       in: query
       description: |
-        A JSON-encoded array of length 2 indicating the narrow for which you'd
-        like to receive events for. For instance, to receive events for the
-        stream `Denmark`, you would specify `narrow=['stream', 'Denmark']`.
-        Another example is `narrow=['is', 'private']` for private messages.
+        A JSON-encoded array of arrays of length 2 indicating the
+        narrow for which you'd like to receive events for. For
+        instance, to receive events for the stream `Denmark`, you
+        would specify `narrow=[['stream', 'Denmark']]`.  Another
+        example is `narrow=[['is', 'private']]` for private messages.
```

Comment:
So, I think it's worse than that; we some time ago migrated to this format for operator/operands as the recommended interface (to allow for `negated`, that is a list of objects with 3 keys: negated/operator/operand.  We should probably just document that; the format documented here is I believe only for backwards-compatibility.

narrow: [{"negated":false,"operator":"stream","operand":268}]

Reply:
The `REQ` call disagrees:

```python
        narrow: NarrowT=REQ(validator=check_list(check_list(check_string, length=2)), default=[]),
```

Reply:
ea8d9efa003643b85d7afc0622048a8519d058bd is relevant here.  I am pretty sure we migrated to dictionaries when tuples only had two values.  I think the way we need negation before this fix was just with a '-' prefix, but I'm not sure about that.

Reply:
ah, one day later, here is negation: 2fb7c0059d36b755c6eed78b95da2c2dbb6517c1

Reply:
Ah—you’re talking about the `narrow` parameter of `get_messages_backend`, while this PR is about the `narrow` parameter of `events_register_backend`.